test/library/packages/Python/doc-examples/parse_docs.py

```diff
@@ -133,6 +140,12 @@ def has_more():
             cur_line += 1
             continue
 
+        # set the noexec flag
+        if test and test_file and lines[cur_line].strip() == "NOEXEC":
+            test.noexec = True
+            cur_line += 1
+            continue
+
```

Comment:
This looks dangerously close to re-inventing sub_test.  Are there functions from sub_test you could be bringing in instead?

Reply:
This does not reinvent sub_test. This is generating tests based on the Python module's documentation, which are then run via sub_test